plan/optimizer.go

```diff
@@ -119,10 +119,10 @@ func doOptimize(flag uint64, logic LogicalPlan, ctx context.Context, allocator *
 	if !AllowCartesianProduct && existsCartesianProduct(logic) {
 		return nil, errors.Trace(ErrCartesianProductUnsupported)
 	}
-	logic.ResolveIndicesAndCorCols()
 	if UseDAGPlanBuilder {
 		return dagPhysicalOptimize(logic)
 	}
+	logic.ResolveIndices()
```

Comment:
There are also some new fields in physical plan that need to resolve. Say the `OutputColumns` in IndexReader. It benefits a lot to resolve after generating physical plan.